### PR TITLE
Feat: Replaced Non-functional Back Button With a Close button(#5982)

### LINKED ISSUE #5982: Replace Non-functional 'Back' Button with 'Cancel' Button on First Panel of 'Create a New Course' Modal
### Issue Description

The `Back` button on the first panel of the `Create a New Course` modal serves no purpose and does not function at all. Since this is the first panel, it should not have a "Back" button. Instead, the button should be replaced with a "Cancel" button that allows the user to exit the modal if needed.

### Steps to Reproduce

1. Open the "Create a New Course" modal.
2. Observe the "Back" button on the first panel.
3. Notice that the "Back" button doesn't work or serve any meaningful purpose.

Refer to the recording below for further details (In case you don't have admin access, which is required to access the button as an instructor or as an admin at some level.):  

https://github.com/user-attachments/assets/b2edcf5c-f1b0-4aff-9960-f86c4a314dc9

### Expected Behaviour

1. The first panel of the "Create a New Course" modal should display a `Cancel` button instead of a "Back" button.
2. The "Back" button should only be present on subsequent panels, as it is irrelevant to have it on the first panel where there is no previous panel screen to return to.

### Screenshots

Screenshot from 2024-10-06 13-29-55

### System Information

 - OS: Ubuntu 22.04.5 LTS
 - Browser: Version 129.0.6668.89 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by Formasitchijoh**
## What this PR does
This PR resolves #5982, by removing the Back button on the first wizard form and replacing it with a Close button

## Screenshots
Before:

https://github.com/user-attachments/assets/7e3b899f-e06b-422c-b974-79001bd78480

https://github.com/user-attachments/assets/519bccfc-f29f-42b1-b570-80bbdd25a77d

After:

https://github.com/user-attachments/assets/d1c739e6-4a15-47fc-ab4d-2415c46f54ef

https://github.com/user-attachments/assets/8c89f107-0a6c-4e25-89ac-8a135207ad21

## Open questions and concerns
@Abishekcs  I wish to ask, most of the abort action button on the form has 'Cancel'. For consistency do you think we should use 'Cancel' in place of 'Close' as you suggested

### CODE REVIEW

**Comment by Abishekcs** [comment]
Yes, @Formasitchijoh, a Cancel button should be used.
Could you please refer to this [updated issue](https://github.com/WikiEducationFoundation/WikiEduDashboard/issues/5982#issuecomment-2396364557)?
Apologies for the earlier mistake.

**Comment by Formasitchijoh** [comment]
Alright @Abishekcs , I have gone though the updated issue I will make the necessary adjustments,

**Comment by Formasitchijoh** [comment]
@Abishekcs @ragesoss   I have update the changes you suggested, Could you check it out and  I was suggesting this back button here be removed since there is no other page it is returning, to
Screenshot 2024-10-10 at 18 18 07

**Comment by Formasitchijoh** [comment]
@ragesoss checking up on this task if there is any feedback i need to update .

**Comment by ragesoss** [comment]
@Abishekcs did you have a look at this?

**Comment by Abishekcs** [comment]
> @Abishekcs did you have a look at this?

yes, but only to verify if it works as intended 😅 , but didn't review the changes made in the code.

File: app/assets/javascripts/components/course_creator/course_creator.jsx [hunk 1]
```diff
@@ -418,7 +419,12 @@ const CourseCreator = createReactClass({
         <label htmlFor="checkbox_id">{I18n.t('courses.creator.copy_courses_with_assignments')}</label>
       </span>
     );
-
+    const { ifadmin: ifAdminStr, wiki_ed: wikiEdStr } = document.getElementById('nav_root').dataset;
+    let isAdminOrInstructor;
+    if (ifAdminStr === 'true' || Features.wikiEd === 'true') {
```
Comment (ragesoss) [inline app/assets/javascripts/components/course_creator/course_creator.jsx:424]: I don't think I understand what's going on here.

File: app/assets/javascripts/components/course_creator/course_form.jsx [hunk 2]
```diff
@@ -193,6 +195,19 @@ const CourseForm = (props) => {
   } else {
     backCondition = props.previous;
   }
+
+  let backOrCancelButton;
+
+  if (props.defaultCourse !== 'ClassroomProgramCourse' || props.isAdminOrInstructor) {
```
[the remark this answers is not in the document]
Comment (Formasitchijoh) [reply app/assets/javascripts/components/course_creator/course_form.jsx:201]: @rageross This checks if the, user is an admin or a  wiki_ed staff

File: app/assets/javascripts/components/course_creator/course_creator.jsx (hunk 1 above)
Comment (Formasitchijoh) [reply app/assets/javascripts/components/course_creator/course_creator.jsx:424]: @rageross This checks if the, user is an admin or a  wiki_ed staff

Comment (Formasitchijoh) [reply app/assets/javascripts/components/course_creator/course_creator.jsx:424]: @rageross This checks if the, user is an admin or a  wiki_ed staff

**Comment by Abishekcs** [comment]
> @Abishekcs @ragesoss I have update the changes you suggested, Could you check it out and I was suggesting this back button here be removed since there is no other page it is returning, to Screenshot 2024-10-10 at 18 18 07

Hi @Formasitchijoh,

 If this is for the `Programs & Events Dashboard`, it should be removed since there's already a cancel button for exiting/close the modal. However, I think it would be better to implement this in a `separate PR`. I also recommend confirming this with @ragesoss first (regarding the back button issue).

**Comment by Formasitchijoh** [comment]
> > @Abishekcs @ragesoss I have update the changes you suggested, Could you check it out and I was suggesting this back button here be removed since there is no other page it is returning, to Screenshot 2024-10-10 at 18 18 07
> 
> Hi @Formasitchijoh,
> 
> If this is for the `Programs & Events Dashboard`, it should be removed since there's already a cancel button for exiting/close the modal. However, I think it would be better to implement this in a `separate PR`. I also recommend confirming this with @ragesoss first (regarding the back button issue).

Thank you @Abishekcs, I will confirm it with him,Please I would equally appreciate your feedback on the code as well, maybe there are somethings i can improve.

**Comment by ragesoss** [comment]
For Programs & Events Dashboard and also Wiki Education Dashboard, I think the 'Back' buttons are currently working as intended *if the user has a previous course that can be cloned*. In those cases, the first panel is the one that gives a choice between 'create a new course' and 'clone a previous course' (with a working Cancel button and no Back button). In that scenario, the Back button works from the Course Form panel; in Programs & Events Dashboard, it returns to the panel for choosing program type, and in Wiki Education Dashboard it returns to the new vs. clone panel.

The bug is for the case where it's Wiki Education Dashboard and the user does not have any courses available to clone, so that the Course Form is the first panel of the course creator. I just checked and it also applies on the Programs & Events Dashboard when there are no available courses to clone: in that case, the program type panel is the one with the non-functional Back button (along with a working Cancel button).

**Comment by Formasitchijoh** [comment]
> For Programs & Events Dashboard and also Wiki Education Dashboard, I think the 'Back' buttons are currently working as intended _if the user has a previous course that can be cloned_. In those cases, the first panel is the one that gives a choice between 'create a new course' and 'clone a previous course' (with a working Cancel button and no Back button). In that scenario, the Back button works from the Course Form panel; in Programs & Events Dashboard, it returns to the panel for choosing program type, and in Wiki Education Dashboard it returns to the new vs. clone panel.
> 
> The bug is for the case where it's Wiki Education Dashboard and the user does not have any courses available to clone, so that the Course Form is the first panel of the course creator. I just checked and it also applies on the Programs & Events Dashboard when there are no available courses to clone: in that case, the program type panel is the one with the non-functional Back button (along with a working Cancel button).

Thank you for the clarification, @ragesoss.
I noticed that the issue you mentioned applies to users with admin or wiki_ed staff roles. However, there are additional concerns for users who are only instructors:

https://github.com/user-attachments/assets/a5927538-391c-45a7-a381-7316117e1079



## Wiki Education Dashboard:
- When an instructor clicks the Create Course button, it leads directly to the first panel of the form.
The Back button in this panel is non-functional.



## Programs and Events Dashboard:
When clicking the Create an Independent Program button, it returns to the panel for choosing the program type.
Similarly , the Back button in this panel is also non-functional.

https://github.com/user-attachments/assets/3c855f21-c375-4690-9f2d-57aece3ce20c

**Comment by ragesoss** [comment]
I think it applies to users with any role, when there are no courses available for that user to clone.

**Comment by Formasitchijoh** [comment]
> I think it applies to users with any role, when there are no courses available for that user to clone.

So  @ragesoss  should the changes i have made be maintained

File: app/assets/javascripts/components/course_creator/course_creator.jsx (hunk 1 above)
Comment (ragesoss) [reply app/assets/javascripts/components/course_creator/course_creator.jsx:424]: `Features.wikiEd` is for whether the server is in `wiki_education` mode or not. It is unrelated to whether the current user is wiki education staff.

File: app/assets/javascripts/components/course_creator/course_creator.jsx [hunk 3]
```diff
@@ -418,7 +419,12 @@ const CourseCreator = createReactClass({
...
+      console.log(ifAdminStr, wikiEdStr, Features.wikiEd);
```
Comment (ragesoss) [inline app/assets/javascripts/components/course_creator/course_creator.jsx:425]: I assume this is for testing?

File: app/assets/javascripts/components/course_creator/course_creator.jsx [hunk 4]
```diff
@@ -418,7 +419,12 @@ const CourseCreator = createReactClass({
         <label htmlFor="checkbox_id">{I18n.t('courses.creator.copy_courses_with_assignments')}</label>
       </span>
     );
-
+    const { ifadmin: ifAdminStr, wiki_ed: wikiEdStr } = document.getElementById('nav_root').dataset;
```
Comment (ragesoss) [inline app/assets/javascripts/components/course_creator/course_creator.jsx:422]: I don't think it should be necessary to pull any data from the html for this. I would expect the fix to be centered on the logic for which panels to show and whether to show the newOrClone panel.

File: app/assets/javascripts/components/course_creator/course_creator.jsx (hunk 3 above)
Comment (Formasitchijoh) [reply app/assets/javascripts/components/course_creator/course_creator.jsx:425]: Oh yes I will remove this.

File: app/assets/javascripts/components/course_creator/course_creator.jsx (hunk 4 above)
Comment (Formasitchijoh) [reply app/assets/javascripts/components/course_creator/course_creator.jsx:422]: The issue is with respect to the non-functional back button for instructors

**Comment by ragesoss** [comment]
Please let change this back to 'ready for review' when you're ready.

**Comment by Formasitchijoh** [comment]
Alright i will make the necessary updates then and get it running,

**Comment by Formasitchijoh** [comment]
> For Programs & Events Dashboard and also Wiki Education Dashboard, I think the 'Back' buttons are currently working as intended _if the user has a previous course that can be cloned_. In those cases, the first panel is the one that gives a choice between 'create a new course' and 'clone a previous course' (with a working Cancel button and no Back button). In that scenario, the Back button works from the Course Form panel; in Programs & Events Dashboard, it returns to the panel for choosing program type, and in Wiki Education Dashboard it returns to the new vs. clone panel.
> 
> The bug is for the case where it's Wiki Education Dashboard and the user does not have any courses available to clone, so that the Course Form is the first panel of the course creator. I just checked and it also applies on the Programs & Events Dashboard when there are no available courses to clone: in that case, the program type panel is the one with the non-functional Back button (along with a working Cancel button).

Hello , @ragesoss  from this clarification i have updated the logic to work as expected, here is how it currently works, i equally updated the display of the back button on the choosing program type page, in Programs & Events Dashboard when the user does not have a cloneable course, `since there wont be a panel for 'create a new course' and 'clone a previous course' . `Note, this  button is available for user with at least clonable course


## User with No clonable course

https://github.com/user-attachments/assets/66b8221c-38ce-45a4-853b-276e9aef8572

## User with clonable course
https://github.com/user-attachments/assets/d0720632-31b2-4dee-9097-659580c53157

**Pull request merged**